packages/backend/src/open_payments/client/service.ts

```diff
@@ -0,0 +1,70 @@
+import axios from 'axios'
+import { OpenAPI, HttpMethod, ValidateFunction } from 'openapi'
+
+import { BaseService } from '../../shared/baseService'
+import { IncomingPaymentJSON } from '../payment/incoming/model'
+
+const REQUEST_TIMEOUT = 5_000 // millseconds
+
+export interface OpenPaymentsClientService {
+  incomingPayment: {
+    get(url: string): Promise<IncomingPaymentJSON | undefined>
+  }
+}
+
+export interface ServiceDependencies extends BaseService {
+  accessToken: string
+  openApi: OpenAPI
+  validateResponse: ValidateFunction<IncomingPaymentJSON>
+}
+
+export async function createOpenPaymentsClientService(
+  deps_: Omit<ServiceDependencies, 'validateResponse'>
+): Promise<OpenPaymentsClientService> {
+  const log = deps_.logger.child({
+    service: 'OpenPaymentsClientService'
+  })
+  const validateResponse =
+    deps_.openApi.createResponseValidator<IncomingPaymentJSON>({
+      path: '/incoming-payments/{incomingPaymentId}',
+      method: HttpMethod.GET
+    })
+  const deps: ServiceDependencies = {
+    ...deps_,
+    logger: log,
+    validateResponse
+  }
+  return {
+    incomingPayment: {
+      get: (url) => getIncomingPayment(deps, url)
+    }
+  }
+}
+
+export async function getIncomingPayment(
```

Comment:
Could we check whether the incoming payment is a resource on the same Rafiki instance and if so, we just use the service to get it instead of the axios call + all the validation?

Reply:
I initially tried that but didn't like having two code paths, one dealing with `IncomingPayments` model and the other dealing with `IncomingPaymentJSON`, the former of which duplicated a lot of functionality in middleware and incoming payments `routes.ts`, including the solution for #598